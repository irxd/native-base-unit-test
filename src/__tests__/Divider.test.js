import React from 'react';
import { render } from '@testing-library/react-native';
import {
  NativeBaseProvider, Divider
} from 'native-base';

describe('<Divider />', () => {
  it('should render divider component properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <Divider testID="my-divider" />
      </NativeBaseProvider>
    );

    expect(queryByTestId('my-divider')).not.toBeNull();
  });

  it('should render divider props properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <Divider testID="my-divider" backgroundColor="black" thickness={2}/>
      </NativeBaseProvider>
    );

    expect(queryByTestId('my-divider').props.style.backgroundColor).toMatch('#000000');
    expect(queryByTestId('my-divider').props.style.height).toEqual(2);
  });
});
