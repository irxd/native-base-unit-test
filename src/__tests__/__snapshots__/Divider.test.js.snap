// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> should render divider props properly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    aria-orientation="horizontal"
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": "#000000",
        "height": 2,
        "width": "100%",
      }
    }
    testID="my-divider"
    thickness={2}
  />
</RNCSafeAreaProvider>
`;
